Extended alerting documentation with information about using Kibana keystore and action types for preconfigured connectors (#65201)
elastic/kibana: docs/user/alerting/pre-configured-connectors.asciidoc

[role="xpack"]
[[pre-configured-action-types-and-connectors]]

== Preconfigured action types and connectors

> You can preconfigure an action type or just a connector to have all the information it needs prior to startup

Reviewer: ```suggestion
You can preconfigure an action type or a connector to have all the information it needs prior to startup
```

- Appear in all spaces because they are not saved objects.
- Cannot be edited or deleted.

Sensitive configuration information, such as credentials, can use the <<creating-keystore, {kib} keystore>>.

> A preconfigured action types has only preconfigured connectors. But preconfigured connectors could belong to the preconfigured action type or to the regular action type as well.

Reviewer: ```suggestion
A preconfigured action types has only preconfigured connectors. Preconfigured connectors can belong to either the preconfigured action type or to the regular action type.
```

[float]
[[preconfigured-connector-example]]
=== Creating a preconfigured connector

The following example shows a valid configuration of two out-of-the box connectors: <<slack-action-type, Slack>> and <<webhook-action-type, Webhook>>.

> You can find all details about available action types configuration <<action-types, here>>.

Reviewer: I don't think this sentence "You can find all details..." is necessary and suggest removing it.

[NOTE]
==============================================
Sensitive properties, such as passwords, can also be stored in the <<creating-keystore, {kib} keystore>>.
==============================================

[float]
[[preconfigured-action-type-example]]

> === Creating a preconfigured action type

Reviewer: This section should come first to match the order of the title "Preconfigured action types and connectors"

> In the `kibana.yml` file:

Reviewer: Suggest removing these lines to make it consistent with the section on preconfigured connectors.  Also the text below the code box explains about xpack.actions.enabled.

Lines to remove:

In the kibana.yml file:

Exclude the action type from xpack.actions.enabledActionTypes.
Add all its preconfigured connectors.

[role="screenshot"]
image::images/pre-configured-connectors-view-screen.png[Pre-configured connector view details]

> If preconfigured connector belong to the preconfigured action type, it is disabled for the details preview.

Reviewer 2: I don't think it matters if the connector is a preconfigured action type.  All preconfigured connectors have their details preview disabled.  So I think this can be shorted to:

> The connector details preview is disabled for preconfigured connectors.

[role="screenshot"]
image::images/pre-configured-action-type-managing.png[Connectors managing tab with pre-cofigured]

[float]
[[managing-pre-configured-action-types]]
=== Managing preconfigured action types

Clicking *Create connector* shows the list of available action types.

> Preconfigured action types are not included because you can't create a connector with a preconfigured action type.

Reviewer 2: I believe the action types aren't included in the list if they are disabled via config, so I think this can be simplified:

> Disabled action types are not included.

[role="screenshot"]

> image::images/pre-configured-action-type-select-type.png[Pre-configured connector create menu]

Reviewer: Suggest cutting off the bottom of this image as its just a lot of white space.

Suggest making the sections from "Alert with a preconfigured connector" on down more of a workflow.  Happy to discuss with you.  We can do this in a separate PR.
